pennylane/numpy/tensor.py

```diff
@@ -17,6 +17,24 @@
 import numpy as onp
 
 from autograd import numpy as _np
+from autograd.extend import primitive, defvjp
+import autograd
+
+# Hotfix since _np.asarray doesn't have a gradient rule defined.

```

Comment:
💯 🎉 

@Thenerdstation this makes me so happy. There are _so many_ places in the codebase where I gritted my teeth and avoided using `np.asarray` because autograd didn't provide gradients.

Could we move this so that it is available via

```python
from pennylane import numpy as np
np.asarray
```

so that users/developers can easily use it?

Reply:
E.g., `default.qubit.autograd` could be sped up by switching to `asarray` as well

Reply:
Want me to add the array -> asarray changes to default.qubit.autograd in this PR or a new PR?

Reply:
I feel like this is in-scope enough if you want to just do it here.

Reply:
👍